### ISSUE TITLE
Change our `relative` server stage to update relative imports instead of converting to absolute.

### ISSUE THREAD

**Issue by flybayer** [blitz-js/blitz#1497]
### What do you want and why?

Currently our [`relative` server stage](https://github.com/blitz-js/blitz/blob/canary/packages/server/src/stages/relative/index.ts) is rewriting relative imports for pages/queries/mutations to be absolute, like this:

```diff
-import {getFoo} from '../bar
+import {getFoo} from 'app/foo/bar
```

But this has some problematic edge cases like https://github.com/blitz-js/blitz/pull/1496 and if a jsconfig or tsconfig doesn't have basePath set.

So instead of converting to an absolute path, just update the relative path as needed for the new location like this

```diff
-import {getFoo} from '../bar
+import {getFoo} from '../../../app/foo/bar
```

**Comment by flybayer** [comment]
Putting in icebox because this should be fixed by this coming change: https://github.com/blitz-js/blitz/discussions/1990

**Comment by flybayer** [comment]
Fixed in #2516
